ios/src/platform_ios.mm

```diff
@@ -57,12 +58,29 @@ bool isContinuousRendering() {
     
 }
 
+std::string setResourceRoot(const char* _path) {
+
+    NSString* path = [NSString stringWithUTF8String:_path];
+
+    if (*_path != '/') {
+        NSString* resources = [[NSBundle mainBundle] resourcePath];
+        path = [resources stringByAppendingPathComponent:path];
+    }
+
+    s_resourceRoot = [path stringByDeletingLastPathComponent];
+
+    return std::string([[path lastPathComponent] UTF8String]);
+
+}
+
 NSString* resolveResourcePath(const char* _path) {
-    
-    NSString* resourcePath = [[[NSBundle mainBundle] resourcePath] stringByAppendingString:@"/"];
-    NSString* internalPath = [NSString stringWithUTF8String:_path];
-    return [resourcePath stringByAppendingString:internalPath];
-    
+
+    if (s_resourceRoot == NULL) {
```

Comment:
Yes, but Tangram::initialize is not called in, say, tests.

On Wed, Sep 23, 2015 at 4:18 PM, Varun <notifications@github.com> wrote:

> In ios/src/platform_ios.mm
> <https://github.com/tangrams/tangram-es/pull/297#discussion_r40252131>:
>
> >  NSString* resolveResourcePath(const char* _path) {
> > -
> > -    NSString* resourcePath = [[[NSBundle mainBundle] resourcePath] stringByAppendingString:@"/"];
> > -    NSString* internalPath = [NSString stringWithUTF8String:_path];
> > -    return [resourcePath stringByAppendingString:internalPath];
> > -
> > +
> > +    if (s_resourceRoot == NULL) {
>
> but its definite that setResourceRoot is called first always, because of
> Tangram::initialize, here
> <https://github.com/tangrams/tangram-es/blob/relative-resources/core/src/tangram.cpp#L38>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/297/files#r40252131>.
>